src/Type.cpp

```diff
@@ -1681,6 +1740,107 @@ Type::get_type_sizeof_string(std::string &s) const
 	s = ss.str();
 }
 
+// If this is C++, create assign operators for volatile structs:
+// Semantically this should work, though for bit fields it's probably not what was intended.
+//volatile struct S0& operator=(const volatile struct S0& val) volatile {
+//	if (this == &val) {
+//		return *this;
+//	}
+//	f0 = val.f0;
+//	f1 = val.f1;
+//	return *this;
+//}
+// As a result of generating this, have to generate 'default' assignment operator as well
+void OutputStructAssignOp(Type* type, std::ostream &out, bool vol)
+{
+	if (CGOptions::lang_cpp()){
+		if (type->has_assign_ops() && (type->eType == eTypeDesc::eStruct)){
```

Comment:
I'll wait for the discussion to settle on using/not using C++11 features in csmith code.